tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -399,3 +476,67 @@ func testAzureSSHConnection(t *testing.T) {
 	err = session.Run("dir")
 	require.NoErrorf(t, err, "failed to communicate vis SSH")
 }
+
+// test if the windows VM is behind the worker node load balancer
+func testAzureVMLoadbalancer(t *testing.T) {
+	ctx := context.Background()
+
+	// create a map to keep track of all the ip configurations associated with the backend pool .
+	ipCSet := make(map[string]struct{})
+	// filler value for map
+	var exists = struct{}{}
```

Comment:
Same comment as before.

Reply:
As mentioned in [comment](https://github.com/openshift/windows-machine-config-bootstrapper/pull/186#discussion_r419704661). 
I am using `map` here as a `set` data structure. 
I am retrieving all the `ipConfig` attached to the backend pool and storing it as a `key` in `map`. 
And later i am getting all the `ipConfig` attached to the Windows VMs and checking if the backend address pool has that ip config attached. [L538](https://github.com/openshift/windows-machine-config-bootstrapper/blob/f6013f4116d675ef5b228b9b014a2f06cc5a9d03/tools/windows-node-installer/test/e2e/azure_test.go#L538)
`map` gives me o(1) loopkup for the keys. for array its o(n)